File: array/11.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        ans = 0
        while l < r:
            ans = max(ans, min(height[l], height[r]) * (r - l))
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return ans


def main():
    a = [2,3,4,5,18,17,6]
    print(Solution().maxArea(a))

if __name__ == "__main__":
    main()

File: leetcode-cpp/11.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        if (height.size() <= 1) return 0;
        int left = 0, right = height.size() - 1;
        int max_area = 0;
        while (left < right) {
            max_area = max(max_area, (right - left) * min(height[left], height[right]));
            if (height[left] < height[right])
                left++;
            else right--;
        }
        return max_area;
    }
};

int main(){
    Solution s;
    vector<int> a({1,8,6,2,5,4,8,3,7});
    cout << s.maxArea(a) <<endl;
    return 0;
}
